lib/pbench/server/filetree.py

```diff
@@ -243,6 +242,59 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def extract(self, path: Path) -> str:
+        """
+        Extract a file from the tarball and return it as a string
+
+        TODO: Should we allow fetching a full directory path, e.g., "/" to
+        extract everything? And, if so, in what form would we return it.
+        Similarly, what about a binary file?
+
+        Args:
+            path: relative path within the tarball of a file
+
+        Raises:
+            MetadataError if an exception occurs unpacking the tarball
+
+        Returns:
+            The named file as a string
+        """
+        try:
+            return (
+                tarfile.open(self.tarball_path, "r:*").extractfile(path).read().decode()
+            )
+        except Exception as exc:
+            raise MetadataError(self.tarball_path, exc)
+
+    def get_metadata(self) -> JSON:
+        """
+        Fetch the values in metadata.log from the tarball, and return a JSON
+        document organizing the metadata by section.
+
+        TODO: Right now this returns just a few particular items that stand
+        out as "likely useful" ... it may make sense to essentially JSONify the
+        entire ConfigParser payload, although there's a lot of data there
+        that's unlikely to be widely useful.
```

Comment:
Great comment to turn into an issue where we can note in the issue that the existing indexing code is also pulling out the `metadata.log` contents and we might want to consider merging that in future work.

Reply:
Covered in #2600. I'm not crazy about spawning so many GitHub issues from this, but I reluctantly accept that our Jira database isn't quite mature enough and hiding them in code isn't the greatest compromise. ☹️